statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/httpfn/HttpFunctionProvider.java

```diff
@@ -41,12 +45,32 @@ public RequestReplyFunction functionOfType(FunctionType type) {
     if (spec == null) {
       throw new IllegalArgumentException("Unsupported type " + type);
     }
-    // specific client reuses the same the connection pool and thread pool
-    // as the sharedClient.
-    OkHttpClient specificClient =
-        sharedClient.newBuilder().callTimeout(spec.maxRequestDuration()).build();
-    RequestReplyClient httpClient =
-        new HttpRequestReplyClient(HttpUrl.get(spec.endpoint()), specificClient);
-    return new RequestReplyFunction(spec.states(), spec.maxNumBatchRequests(), httpClient);
+    return new RequestReplyFunction(
+        spec.states(), spec.maxNumBatchRequests(), buildHttpClient(spec));
+  }
+
+  private RequestReplyClient buildHttpClient(HttpFunctionSpec spec) {
+    // We need to build a UDS HTTP client
+    if (spec.unixDomainSocket() != null) {
+      OkHttpClient specificClient =
+          sharedClient
+              .newBuilder()
+              .socketFactory(new AFUNIXSocketFactory.FactoryArg(spec.unixDomainSocket()))
```

Comment:
Is there a reason that you chose not to implement a custom UDS socket factory like in the example at https://github.com/square/okhttp/tree/master/samples/unixdomainsockets/src/main/java/okhttp3/unixdomainsockets?

I'm asking only because we should try to have the `statefun-flink-core` dependencies as slim as possible, as it directly influences the size of our base distributions. It might also not be too big of a deal if the `junixsocket-core` library is small.

Reply:
I didn't used the uds library proposed by okhttp `jnr-unixsocket` because it doesn't look in a very good shape (last release is 0.31 https://github.com/jnr/jnr-unixsocket/releases)

I'll try to find out if we can use only `junixsocket-core`

Reply:
Ok so this factory comes from `common` module, that is the least module we can use because it's the one that contains the `java.net.Socket` implementation